### PR TITLE
Change matches function to include all games for a competition season

### PR DESCRIPTION
**Pull request by scottyj779**
It's been pointed out that the `sb.matches()` function only returns the games we have already collected for the competition season even though the API returns all the matches for the comp-season regardless of whether they have been collected or not. 

StatsBombR also returns all matches so better to have the two aligned.

I also removed the matches test for dual managers as the particular instance has been fixed in the raw data so that test fails now.

### CODE REVIEW

**Review by WillTGM — APPROVED** [review]
Looks good. We probably want to publish some release notes to cover the change as it _might_ affect user workflows.

File: tests/statsbombpy_test/test_sb.py [hunk 1]
```diff
@@ -37,10 +37,6 @@ def test_matches(self):
         self.assertIsInstance(matches, pd.DataFrame)
 
         matches = sb.matches(competition_id=11, season_id=1)
-        self.assertEquals(
-            matches.query("match_id == 9695")["home_managers"].iloc[0],
-            "Juan Rubén Uría Corral, Marcelino García Toral",
-        )
```
Comment (FranGoitia) [inline tests/statsbombpy_test/test_sb.py:43]: Why did you remove this test ?

Comment (scottyj779) [reply tests/statsbombpy_test/test_sb.py]: That test failed because they changed the underlying data to only have one manager. I have pushed another commit to replace the test with another match where there's two managers.
